concordium-consensus/src/Concordium/GlobalState/Persistent/Account/StructureV1.hs

```diff
@@ -0,0 +1,1314 @@
+-- This is because we typically require ready access to the locked balance so that we can compute
+-- the available balance.
+--
+-- The stake is not stored under a reference (excepting the baker info, as per the definition of
+-- 'PersistentAccountStakeEnduring'). This is since the information is relatively succinct.
+data PersistentAccountEnduringData (av :: AccountVersion) = PersistentAccountEnduringData
+    { -- |The Merkle hash computed from the other fields.
+      paedHash :: !(AccountMerkleHash av),
+      -- |A reference to the persisting account data.
+      paedPersistingData :: !(EagerBufferedRef PersistingAccountData),
+      -- |The encrypted amount. Invariant: if this is present, it will not satisfy
+      -- 'isZeroAccountEncryptedAmount'.
+      paedEncryptedAmount :: !(Nullable (LazyBufferedRef PersistentAccountEncryptedAmount)),
+      -- |The release schedule and total locked amount. Invariant: if this is present,
+      -- there will be scheduled releases, and the amount will be the total of them.
+      paedReleaseSchedule :: !(Nullable (LazyBufferedRef AccountReleaseSchedule, Amount)),
+      -- |The staking details associated with the account.
+      paedStake :: !(PersistentAccountStakeEnduring av)
+    }
+
+-- |Get the locked amount from a 'PersistingAccountEnduringData'.
+paedLockedAmount :: PersistentAccountEnduringData av -> Amount
+paedLockedAmount PersistentAccountEnduringData{..} = case paedReleaseSchedule of
+    Some (_, amt) -> amt
+    Null -> 0
+
+instance HashableTo (AccountMerkleHash av) (PersistentAccountEnduringData av) where
+    getHash = paedHash
+
+-- |Construct a 'PersistentAccountEnduringData' from the components by computing the hash.
+makeAccountEnduringData ::
+    ( MonadBlobStore m
+    ) =>
+    EagerBufferedRef PersistingAccountData ->
+    Nullable (LazyBufferedRef PersistentAccountEncryptedAmount) ->
+    Nullable (LazyBufferedRef AccountReleaseSchedule, Amount) ->
+    PersistentAccountStakeEnduring 'AccountV2 ->
+    m (PersistentAccountEnduringData 'AccountV2)
+makeAccountEnduringData paedPersistingData paedEncryptedAmount paedReleaseSchedule paedStake = do
+    amhi2PersistingAccountDataHash <- getHashM paedPersistingData
+    (amhi2AccountStakeHash :: AccountStakeHash 'AccountV2) <- getHashM paedStake
+    amhi2EncryptedAmountHash <- case paedEncryptedAmount of
+        Null -> return initialAccountEncryptedAmountHash
+        Some e -> getHash <$> (loadPersistentAccountEncryptedAmount =<< refLoad e)
+    amhi2AccountReleaseScheduleHash <- case paedReleaseSchedule of
+        Null -> return Transient.emptyAccountReleaseScheduleHash
+        Some (rs, _) -> getHashM rs
+    let hashInputs :: AccountMerkleHashInputs 'AccountV2
+        hashInputs = AccountMerkleHashInputsV2{..}
+        !paedHash = getHash hashInputs
+    return $! PersistentAccountEnduringData{..}
+
+-- |[For internal use.] Recompute the Merkle hash of the enduring account data.
```

Comment:
Please elaborate what "for internal use" means. What is the scope of "internal"? And why?